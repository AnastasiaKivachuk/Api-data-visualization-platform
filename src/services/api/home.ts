import { gql } from '@apollo/client';
import client from '../../../apollo-client';

export const getHome = () => client.query({
  query: gql`
        query HomeScheme {
            company {
                ceo
                employees
                founded
                founder
                links {
                    elon_twitter
                    website
                }
                name
                summary
            }
            ships {
                id
                image
                model
                name
                url
                year_built
            }
            rockets {
                wikipedia
                name
                id
                description
            }
        }
    `,
});
