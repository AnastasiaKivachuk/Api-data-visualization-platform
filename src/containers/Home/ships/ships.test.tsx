import {shallow} from "enzyme";
import React from "react";

import {ShipsInterface} from "../../../interfaces/home.interface";
import Ships from "./index";

const MockShipsData: ShipsInterface[] = [{
    id: 'NRCQUEST',
    image: 'https://i.imgur.com/GvtuYUN.jpg',
    model: 'Marmac 303',
    name: 'NRC Quest',
    url: 'https://www.marinetraffic.com/en/ais/details/ships/shipid:447294/vessel:NRC%20QUEST',
    year_built: 2002
}];

const setUp = (props: { ships: ShipsInterface[] } | null) => shallow(<Ships {...props} />);

describe("Ships component", () => {

    it("should render Ships component", () => {
        const component = setUp({ships: MockShipsData})
        expect(component).toMatchSnapshot();
    })

    it("check all elements with props", () => {
        const component = setUp({ships: MockShipsData})
        expect(component.find('ForwardRef(Typography)').text()).toEqual('Ships');
        expect(component.find('ShipCard').length).toBe(1)
    })

    it("check all elements with defaultProps", () => {
        const component = setUp(null)
        expect(component.find('ShipCard').length).toBe(0)
    })
})
