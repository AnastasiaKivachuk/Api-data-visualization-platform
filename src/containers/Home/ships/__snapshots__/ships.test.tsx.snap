// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Ships component should render Ships component 1`] = `
<Styled(div)>
  <ForwardRef(Typography)
    gutterBottom={true}
    textAlign="center"
    variant="h4"
  >
    Ships
  </ForwardRef(Typography)>
  <ForwardRef(Grid)
    container={true}
    gap="20px"
    justifyContent="center"
  >
    <ShipCard
      id="NRCQUEST"
      image="https://i.imgur.com/GvtuYUN.jpg"
      key="NRCQUEST"
      model="Marmac 303"
      name="NRC Quest"
      url="https://www.marinetraffic.com/en/ais/details/ships/shipid:447294/vessel:NRC%20QUEST"
      year_built={2002}
    />
  </ForwardRef(Grid)>
</Styled(div)>
`;
